BuildAddOn.py

```diff
@@ -173,34 +174,40 @@ def GetProjectGenerationParams (workspaceRootFolder, buildPath, platformName, de
 
     if optionalParams is not None:
         for key in optionalParams:
-            projGenParams.append (f'-D{key}={optionalParams[key]}')
+            if optionalParams[key] is None:
+                projGenParams.append (f'-D{key}')
+            else:
+                projGenParams.append (f'-D{key}={optionalParams[key]}')
 
     projGenParams.append (str (workspaceRootFolder))
 
     return projGenParams
 
 
-def BuildAddOn (configData, platformName, workspaceRootFolder, buildFolder, devKitFolder, version, configuration, languageCode=None):
-    addOnName = configData['addOnName']
-    optionalParams = None
-    if 'addOnSpecificCMakeParameterEnvVars' in configData:
-        optionalParams = {}
-        for key in configData['addOnSpecificCMakeParameterEnvVars']:
-            value = os.environ.get (key)
-            if value is None:
-                raise Exception (f'{key} - Environment variable not found!')
-            optionalParams[key] = value
+def ParseCMakeCache (buildPath):
+    addOnName = None
+    for line in open (buildPath / 'CMakeCache.txt'):
+        if 'AC_ADDON_NAME' in line:
+            addOnName = line.strip ().split ('=')[1]
 
-    buildPath = buildFolder / addOnName / version
+    return addOnName
+
+
+def BuildAddOn (optionalParams, platformName, workspaceRootFolder, buildFolder, devKitFolder, version, configuration, languageCode=None):
+    buildPath = buildFolder / 'TempBuildDir' / version
     if languageCode is not None:
         buildPath = buildPath / languageCode
 
     # Add params to configure cmake
-    projGenParams = GetProjectGenerationParams (workspaceRootFolder, buildPath, platformName, devKitFolder, version, languageCode, optionalParams)
+    projGenParams = GetProjectGenerationParams (optionalParams, workspaceRootFolder, buildPath, platformName, devKitFolder, version, languageCode)
     projGenResult = subprocess.call (projGenParams)
     if projGenResult != 0:
         raise Exception ('Failed to generate project!')
     
+    addOnName = ParseCMakeCache (buildPath)

```

Comment:
Why don't we just add this name as a parameter to the build script? This way it could be passed to cmake, and we won't need these rounds with the cmake cache file.